c-api/compat-5.3.c

```diff
@@ -350,36 +369,37 @@ COMPAT53_API void luaL_traceback (lua_State *L, lua_State *L1,
 COMPAT53_API int luaL_fileresult (lua_State *L, int stat, const char *fname) {
   const char* s = NULL;
   int en = errno;  /* calls to Lua API may change this value */
+#if defined(COMPAT53_HAVE_STRERROR_R) || defined(COMPAT53_HAVE_STRERROR_S)
+  char buf[512] = { 0 };
+#endif // buffer for threadsafe variants of strerror if possible
   if (stat) {
     lua_pushboolean(L, 1);
     return 1;
   }
   else {
     lua_pushnil(L);
-#if defined(__GLIBC__) || defined(_POSIX_VERSION) || defined(__APPLE__) || (!defined (__MINGW32__) && defined(__GNUC__) && (__GNUC__ < 6))
+#if defined(COMPAT53_HAVE_STRERROR_R)
```

Comment:
Drop the `defined`

Reply:
`#if COMPAT53_HAVE_STRERROR_R` errors if the `COMPAT53_HAVE_STRERROR_R` is undefined: I don't define it when it doesn't exist. Should I define it to 0 in the `#else` branch?

Reply:
Generally I do it like this:
```C
#ifndef HAVE_FOO
#define HAVE_FOO (_WIN32)
#endif
```

However there's a new warning in gcc 7 (`-Wexpansion-to-defined`) which warns on the above style if you use e.g. `defined(__GLIBC__)` where I used `_WIN32`, so yes: in some cases you define to `0`.